build_r.R

```diff
@@ -17,12 +17,42 @@ INSTALL_AFTER_BUILD <- !("--skip-install" %in% args)
 }
 
 # system() will not raise an R exception if the process called
-# fails. Wrapping it here to get that behavior
-.run_shell_command <- function(cmd, ...) {
-    exit_code <- system(cmd, ...)
-    if (exit_code != 0L) {
+# fails. Wrapping it here to get that behavior.
+#
+# system() introduces a lot of overhead, at least on Windows,
+# so trying processx if it is available
+.run_shell_command <- function(cmd, args, strict = TRUE) {
```

Comment:
Is it possible to import this function from `R-package/src/install.libs.R` somehow? I assume this script `build_r.R` will be never run without content from this repo or this use case is weird  at least.

Reply:
There is not an R equivalent of importing one function from a module the way you can in Python.

R has the concept of running ALL of a single file with  `source()`, and I did consider that. But I think we should avoid it. I would rather have this code duplicated in two places than introduce more indirection and complexity.

We have CI to catch issues, so I think `.run_shell_command()`  is a case where copying code is better than having shared code.

Reply:
> There is not an R equivalent of importing one function from a module the way you can in Python.

Okay 🙁